lib/compress.js

```diff
@@ -706,9 +706,9 @@ merge(Compressor.prototype, {
         return x;
     };
 
-    function is_iife_call(node) {
+    function is_iife_call(node, traditional) {
         if (node instanceof AST_Call && !(node instanceof AST_New)) {
-            return is_func_expr(node.expression) || is_iife_call(node.expression);
+            return is_func_expr(node.expression, traditional) || is_iife_call(node.expression);
```

Comment:
Thanks - it was originally `node.expression instanceof AST_Function`, but I made a blanket replacement in https://github.com/mishoo/UglifyJS2/pull/2170/files#diff-230c53e13e3ee367aec7d49470eb42caR714 without realising `AST_Arrow` doesn't benefit from `()` :arrow_right: `!`